### PR TITLE
inter-nerdfont: init at 4.0

### PR DESCRIPTION
**Pull request by midischwarz12**
## Description of changes

Adding NerdFont patch of Inter to nixpkgs.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/in/inter-nerdfont/package.nix
Comment (Sigmanificient) [inline pkgs/by-name/in/inter-nerdfont/package.nix:10]: Use `finalAttrs` to allow versison overrides
```suggestion
stdenvNoCC.mkDerivation (finalAttrs: {
  pname = "inter-nerdfont";
  version = "4.0";

  src = fetchzip {
    url = "https://github.com/rsms/inter/releases/download/v${finalAttrs.version}/Inter-${finalAttrs.version}.zip";
```

(with extra parenthesis at the end)

File: pkgs/by-name/in/inter-nerdfont/package.nix
Comment (Sigmanificient) [inline pkgs/by-name/in/inter-nerdfont/package.nix:30]: Can you import dependencies directly as arguments?
```suggestion
  buildInputs = [ fontforge nerd-font-patcher ];
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/by-name/in/inter-nerdfont/package.nix
Comment (cvoges12) [reply pkgs/by-name/in/inter-nerdfont/package.nix:28]: @Sigmanificient Why is the `pkgs.` prefix un-needed?

File: pkgs/by-name/in/inter-nerdfont/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/in/inter-nerdfont/package.nix:1]:
```suggestion
{ lib, fontforge, nerd-font-patcher, stdenvNoCC, fetchzip }:
```

File: pkgs/by-name/in/inter-nerdfont/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/in/inter-nerdfont/package.nix:18]:
```suggestion
  buildPhase = ''
    runHook preBuild

    nerd-font-patcher Inter.ttc
    
    runHook postBuild
'';

  installPhase = ''
    runHook preInstall

    install -Dm444 'Inter Nerd Font.ttc' $out/share/fonts/truetype/InterNerdFont.ttc
```

Comment (eclairevoyant) [reply pkgs/by-name/in/inter-nerdfont/package.nix:18]: probably could also use `dontConfigure = true;`

File: pkgs/top-level/all-packages.nix
Comment (eclairevoyant) [inline pkgs/top-level/all-packages.nix:28935]:
```suggestion
```

this is unneeded with by-name

File: pkgs/by-name/in/inter-nerdfont/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/in/inter-nerdfont/package.nix:32]:
```suggestion
    description = "NerdFont patch of the Inter font";
```

**Review by eclairevoyant — DISMISSED** [review]
Thanks for the contribution! I have some additional feedback below.

Also, please rebase + squash into two commits:
* `maintainers: add cvoges12`
* `inter-nerdfont: init at 4.0`

File: pkgs/by-name/in/inter-nerdfont/package.nix
Comment (eclairevoyant) [reply pkgs/by-name/in/inter-nerdfont/package.nix:28]: you'd have to change the package expression's argset as well

**Comment by cvoges12** [comment]
> Thanks for the contribution! I have some additional feedback below.
> 
> Also, please rebase + squash into two commits:
> 
> * `maintainers: add cvoges12`
> * `inter-nerdfont: init at 4.0`

Wouldn't I need to send in a new PR for this? @eclairevoyant

**Comment by eclairevoyant** [comment]
No, you need to use `git rebase -i`

**Comment by cvoges12** [comment]
@eclairevoyant I had some trouble because of needing to split commits. But I managed to rebase in the end. It is ready to merge now.

**Comment by cvoges12** [comment]
@eclairevoyant Is there anything this is waiting on before merging?

**Comment by cvoges12** [comment]
@eclairevoyant @Sigmanificient Is there anything needed to get this merged?

**Comment by midirhee12** [comment]
Okay, my bad. This seemed to close itself after doing a force push. I'll have to recreate this PR.

**Comment by midirhee12** [comment]
Reopening.

**Comment by eclairevoyant** [comment]
Locking due to mass ping; please feel free to open a new PR.

**Pull request closed without merging**
